packages/renderer/src/lib/preferences/PreferencesConnectionDetailsTerminal.spec.ts

```diff
@@ -0,0 +1,206 @@
+/**********************************************************************
+ * Copyright (C) 2024 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import '@testing-library/jest-dom/vitest';
+
+import { render, screen, waitFor } from '@testing-library/svelte';
+import { get } from 'svelte/store';
+import { beforeEach, expect, test, vi } from 'vitest';
+
+import { providerTerminals } from '/@/stores/provider-terminal-store';
+import type {
+  ProviderContainerConnectionInfo,
+  ProviderInfo,
+  ProviderKubernetesConnectionInfo,
+} from '/@api/provider-info';
+
+import PreferencesConnectionDetailsTerminal from './PreferencesConnectionDetailsTerminal.svelte';
+
+const getConfigurationValueMock = vi.fn();
+const shellInProviderConnectionMock = vi.fn();
+const shellInProviderConnectionResizeMock = vi.fn();
+const shellInProviderConnectionCloseMock = vi.fn();
+const receiveEndCallbackMock = vi.fn();
+
+vi.mock('xterm', () => {
+  return {
+    Terminal: vi
+      .fn()
+      .mockReturnValue({ loadAddon: vi.fn(), open: vi.fn(), write: vi.fn(), dispose: vi.fn(), onData: vi.fn() }),
+  };
+});
+
+beforeEach(() => {
+  vi.resetAllMocks();
+  (window as any).getConfigurationValue = getConfigurationValueMock;
```

Comment:
Object.defineProperty is to be preferred to prevent use of any. And if you use any, you should have an eslint directive to allow it